apps/dashboard/src/components/workflow-editor/workflow-provider.tsx

```diff
@@ -105,9 +111,13 @@ export const WorkflowProvider = ({ children }: { children: ReactNode }) => {
   const isUpdatePatchPending = isPatchPending || isUpdatePending || hasPendingItems;
 
   const update = useCallback(
-    (data: UpdateWorkflowDto) => {
+    ({ data, onSuccess }: { data: UpdateWorkflowDto; onSuccess?: (workflow: WorkflowResponseDto) => void }) => {
```

Comment:
I think it would be better DX to do 
``` const update = useCallback(
    (data: UpdateWorkflowDto, { onSuccess }: { onSuccess?: (workflow: WorkflowResponseDto) => void }) => {```

It's also a safer change and doesn't introduce the super generic data key across the board.